neqo-transport/src/connection.rs

```diff
@@ -1228,10 +1228,23 @@ impl Connection {
                     rs.inbound_stream_frame(fin, offset, data)?;
                 }
             }
-            Frame::MaxData { maximum_data } => self
-                .flow_mgr
-                .borrow_mut()
-                .conn_increase_max_credit(maximum_data),
+            Frame::MaxData { maximum_data } => {
+                let conn_was_blocked = self.flow_mgr.borrow().conn_credit_avail() == 0;
+                if self
+                    .flow_mgr
+                    .borrow_mut()
+                    .conn_increase_max_credit(maximum_data)
+                    && conn_was_blocked
```

Comment:
We should short-circuit on conn_was_blocked by putting that first.  The compiler might do this, but probably not.

Reply:
can't do that because conn_increase_max_credit() actually does something. I think I'll add a local variable to make this clearer.